src/main/java/com/statsig/androidsdk/StatsigNetwork.kt

```diff
@@ -27,227 +26,186 @@ private val RETRY_CODES: IntArray = intArrayOf(
+            response
+        } catch (_ : Exception) {
+            null
         }
+    }
 
-        fun apiPostLogs(
-            api: String,
-            sdkKey: String,
-            bodyString: String,
-            sharedPrefs: SharedPreferences?,
-        ) {
-            coroutineScope.launch(Dispatchers.IO) {
-                postRequestAsync(
-                    api,
-                    LOGGING_ENDPOINT,
-                    sdkKey,
-                    bodyString,
-                    LogEventResponse::class,
-                    3,
-                    sharedPrefs,
+    override fun pollForChanges(
+        api: String,
+        sdkKey: String,
+        user: StatsigUser?,
+        metadata: StatsigMetadata
+    ): Flow<InitializeResponse?> {
+        @Suppress("RemoveExplicitTypeArguments") // This is needed for tests
+        return flow<InitializeResponse?> {
+            while (true) {
+                delay(POLLING_INTERVAL_MS) // If coroutine is cancelled, this delay will exit the while loop
+                val body = mapOf(
+                    USER to user,
+                    STATSIG_METADATA to metadata,
+                    LAST_SYNC_TIME_FOR_USER to lastSyncTimeForUser
                 )
+                emit(postRequest(api, INITIALIZE_ENDPOINT, sdkKey, gson.toJson(body), 0, null))
             }
         }
+    }
 
-        @Synchronized
-        fun apiRetryFailedLogs(
-            api: String,
-            sdkKey: String,
-            sharedPrefs: SharedPreferences,
-        ) {
-            val savedLogs = getSavedLogs(sharedPrefs)
-            if (savedLogs == null || savedLogs.isEmpty()) {
-                return
-            }
-            sharedPrefs.edit().remove(OFFLINE_LOGS_KEY).apply()
-            savedLogs.map {
-                apiPostLogs(api, sdkKey, it.requestBody, sharedPrefs)
-            }
-        }
+    override suspend fun apiPostLogs(api: String, sdkKey: String, bodyString: String, sharedPrefs: SharedPreferences) {
+        postRequest<LogEventResponse>(api, LOGGING_ENDPOINT, sdkKey, bodyString, 3, sharedPrefs)
+    }
 
-        private fun addFailedLogRequest(sharedPrefs: SharedPreferences, requestBody: String) {
-            var savedLogs = getSavedLogs(sharedPrefs)
-            if (savedLogs == null) {
-                savedLogs = ArrayList<StatsigOfflineRequest>()
-            }
-            savedLogs.add(StatsigOfflineRequest(java.lang.System.currentTimeMillis(), requestBody))
-            saveFailedRequests(sharedPrefs, StatsigPendingRequests(ArrayList(savedLogs)))
+    override suspend fun apiRetryFailedLogs(api: String, sdkKey: String, sharedPrefs: SharedPreferences) {
+        val savedLogs = sharedPrefs.getSavedLogs()
+        if (savedLogs.isEmpty()) {
+            return
         }
+        sharedPrefs.edit { remove(OFFLINE_LOGS_KEY) }
+        savedLogs.forEach { apiPostLogs(api, sdkKey, it.requestBody, sharedPrefs) }
```

Comment:
There may be benefits in posting all saved logs concurrently. Didn't implement this, but would speed up initialization.

Reply:
We can make this fire and forget so initialize doesnt wait